cmake/modules/gluecodium/gluecodium/Generate.cmake

```diff
@@ -182,8 +182,15 @@ function(apigen_generate)
     GENERATOR ${APIGEN_GENERATOR}
     BUILD_OUTPUT_DIR "${APIGEN_BUILD_OUTPUT_DIR}")
 
-  set (_lime_interface_sources "$<FILTER:$<TARGET_PROPERTY:${APIGEN_TARGET},INTERFACE_SOURCES>,INCLUDE,.*\\.lime$>")
-  set (_lime_sources "$<FILTER:$<TARGET_PROPERTY:${APIGEN_TARGET},SOURCES>,INCLUDE,.*\\.lime$>")
+  if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
+    set (_lime_interface_sources "$<FILTER:$<TARGET_PROPERTY:${APIGEN_TARGET},INTERFACE_SOURCES>,INCLUDE,.*\\.lime$>")
+    set (_lime_sources "$<FILTER:$<TARGET_PROPERTY:${APIGEN_TARGET},SOURCES>,INCLUDE,.*\\.lime$>")
+    set (_lime_dependencies ${_lime_interface_sources} ${_lime_sources})
+  else()
+    message (WARNING "CMake < 3.15 is obsolete and may work wrongly with Gluecodium")
+    set (_lime_interface_sources "$<TARGET_PROPERTY:${APIGEN_TARGET},INTERFACE_SOURCES>")
+    set (_lime_sources "$<TARGET_PROPERTY:${APIGEN_TARGET},SOURCES>")
```

Comment:
Why this branch is not setting `_lime_dependencies` var? I think it should. Or, better yet, set the var outside of the `if` below.

Reply:
Because for CMake  < 3.15 this variable must be empty, otherwise custom command will depend on other (than lime) sources as well, but it shouldn't